tests/DoctrineObjectTest.php

```diff
@@ -2866,4 +2927,58 @@ public function testNestedHydrationByReference(): void
         $this->assertSame('value', $entity->getToOne(false)->getField(false));
         $this->assertSame('2019-01-24 12:00:00', $entity->getCreatedAt()->format('Y-m-d H:i:s'));
     }
+
+    public function testHandleEnumConversionUsingByValue(): void
+    {
+        if (PHP_VERSION_ID < 80100) {
+            $this->markTestSkipped('PHP 8.1 required for enum compatibility');
+        }
+
+        // When using hydration by value, it will use the public API of the entity to set values (setters)
+        $entity = new Assets\SimpleEntityWithEnum();
+        $this->configureObjectManagerForSimpleEntityWithEnum();
+
+        $value = 1;
+        $data  = ['enum' => $value];
+
+        $this->hydratorByValue->addStrategy('enum', new Assets\SimpleEnumStrategy());
+        $entity = $this->hydratorByValue->hydrate($data, $entity);
+
+        $this->assertInstanceOf(SimpleEnum::class, $entity->getEnum());
+        $this->assertEquals(SimpleEnum::tryFrom($value), $entity->getEnum());
+    }
+
+    public function testNullValueIsNotConvertedToEnum(): void
+    {
+        if (PHP_VERSION_ID < 80100) {
+            $this->markTestSkipped('PHP 8.1 required for enum compatibility');
+        }
+
+        $entity = new Assets\SimpleEntityWithEnum();
+        $this->configureObjectManagerForSimpleEntityWithEnum();
+
+        $data = ['enum' => null];
+
+        $this->hydratorByValue->addStrategy('enum', new Assets\SimpleEnumStrategy());
+        $entity = $this->hydratorByValue->hydrate($data, $entity);
+
+        $this->assertNull($entity->getEnum());
+    }
+
+    public function testWrongEnumBackedValueThrowsException(): void
+    {
+        if (PHP_VERSION_ID < 80100) {
```

Comment:
I think PHPUnit has `@requires php >= 8.1` or such?

See https://phpunit.readthedocs.io/en/9.5/incomplete-and-skipped-tests.html#incomplete-and-skipped-tests-requires-tables-api